### PR TITLE
🔢 Add optional revision number to flow refs

### PR DESCRIPTION
**Pull request by rowanseymour**
Note that this will be the revision number when the run started

### CODE REVIEW

File: flows/actions/testdata/call_classifier.json [hunk 1]
```diff
@@ -163,11 +163,11 @@
                     {
                         "url": "https://api.wit.ai/message?v=20200513&q=Hi+everybody",
                         "status_code": 200,
-                        "status": "success",
```
Comment (rowanseymour) [inline flows/actions/testdata/call_classifier.json:166]: some annoying unrelated ordering changes in the diff here...

File: flows/actions/enter_flow.go [hunk 2]
```diff
@@ -55,11 +55,11 @@ func (a *EnterFlowAction) Execute(run flows.Run, step flows.Step, logModifier fl
...
 
 	if run.Session().Type() != flow.Type() {
-		a.fail(run, errors.Errorf("can't enter %s of type %s from type %s", flow.Reference(), flow.Type(), run.Session().Type()), logEvent)
+		a.fail(run, errors.Errorf("can't enter %s of type %s from type %s", flow.Reference(false), flow.Type(), run.Session().Type()), logEvent)
 		return nil
 	}
 
 	run.Session().PushFlow(flow, run, a.Terminal)
-	logEvent(events.NewFlowEntered(a.Flow, run.UUID(), a.Terminal))
+	logEvent(events.NewFlowEntered(flow.Reference(false), run.UUID(), a.Terminal))
```
Comment (rowanseymour) [inline flows/actions/enter_flow.go:63]: by getting the reference from the flow instead of using the reference in the flow, we get the latest flow name

**Review by norkans7 — APPROVED** [review]
